z390test/src/test/groovy/org/z390/test/RunCmdProc.groovy

```diff
@@ -0,0 +1,89 @@
+package org.z390.test
+
+import org.junit.jupiter.api.Test
+
+class RunCmdProc extends z390Test {
+    /*
+    Because the source contains hard coded file paths, we need to replace these with a path using basePath method.
+    We load the source into a variable and the update the file paths using a series of regex replacements
```

Comment:
Change "... and the update ..." to "... and then update ...".

This test is a tough one! Impressive work.

Reply:
Update. Even though I corrected a typo in the comment above, I apparently did not read the text! The problem is in the hard coded filenames. I have two additional ways for me to fix this, avoiding Groovy requiring "inside knowledge" of the assembler source.

1. This one is easy for me: change the filenames to begin with ".\" (Windows) and "./" (Linux/Mac). For TESTCMD3.MLC change them to
    TestCmdW DC    C'.\rt\test\bat\TESTCMD3.BAT',X'00'   Windows command
    TestCmdL DC    C'./rt/test/bash/testcmd3',X'00' Linux/Mac command
Similar changes for TESTCMD4.MLC.
Running bash/runcmdproc on Linux and RUNCMDPROC.BAT on Windows works with these changes.
Could you modify your test to do just these changes this and try it? Since I do not claim any knowledge of Groovy, I do not know if the test will work. If it works, then I will submit a PR for the changes. That way,the Groovy test will be simple: keep your hack to set COMSPEC and just use asmlg as in the bash runcmdproc script and the bat RUNCMDPROC.BAT file. If it does not work, the next idea may work.

2. Assuming item 1 fails, we could add a new "env" for something like you did in the runrtbsam and runrtqsam tests:
        env = ['CMD3DIR': basePathRelative('rt', 'test')]
I'm guessing here what the relative path is. I don't know if basePathRelative allows the path to be a directory, so this may not work. If this is allowed, I can modify TESTCMD3.MLC to extract this from the environment variable and modify the file name accordingly. I will need to know if the resulting environment variable ends with the file separator character.

Reply:
I'll read this later and let you know my thoughts.. open to ideas on this because it was a PITA handling the filenames.

Reply:
OK.. So I think I don't like option 1 above because it assumes that the z390 source and binary is in a specific location relative to the current directory. The Environment variable `Z390_PROJECT_ROOT` does have a default that assumes this but I don't want the tests to build this assumption in just in case someone has their source on their machine setup differently. 
For this reason, I do prefer option 2. Let the script pass the filename. This does not require special handling because the basepath and basepathrelative functions will use the correct slash depending on the OS that the test is being run with.. So no special handling would be required in the test to support both which is what we want.. 
I generally like full paths for filenames but when replacing the filename in the assembly, it was too long, so I created the basepathrelative function that provides a relative link which is shorter. I am assuming that if you go the envvar path that we would not be updating the source and that there would not be a restriction on the filename length.. 
Hope this makes sense.

Reply:
@jyganci Lets work together on a way to fix this one.. would love to remove the process of copying and modifying the source to get the tests to run..

Reply:
@jyganci -- after our discussion, I think the decision was to leave this as is and depending on who goes first in terms of merges, we will deal with the update to pass the file names to the test modules via envvars. Let me know if I got this correct